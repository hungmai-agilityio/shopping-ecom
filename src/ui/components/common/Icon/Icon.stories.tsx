import { Meta, StoryObj } from '@storybook/react';

// Components
import Icon from '@/ui/components/common/Icon';
import { TYPE } from '@/constants';

const meta: Meta<typeof Icon> = {
  component: Icon,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  },
  argTypes: {
    src: {
      description: 'Image to render'
    },
    alt: {
      description: 'Describes the image and shows if the image is corrupted'
    },
    width: {
      description: 'Width for images'
    },
    height: {
      description: 'Height for images'
    },
    variant: {
      description: 'Set type optional for icon',
      control: 'radio',
      options: Object.values(TYPE)
    },
    onClick: {
      description: 'Handle the event when clicking on the button'
    },
    isDisabled: {
      description: 'Set disabled icon'
    },
    styles: {
      description: 'Set styles for icon'
    }
  }
};

export default meta;

type Story = StoryObj<typeof Icon>;

export const basic: Story = {
  args: {
    src: '/cart.svg',
    alt: 'Cart Icon',
    width: 24,
    height: 24,
    onClick: () => {
      alert(123);
    }
  }
};

export const hasBorder: Story = {
  args: {
    src: '/cart.svg',
    alt: 'Cart Icon',
    width: 24,
    height: 24,
    onClick: () => {
      alert(123);
    },
    variant: TYPE.SECOND
  }
};

export const notAction: Story = {
  args: {
    src: '/heart.svg',
    alt: 'Cart Icon',
    width: 30,
    height: 30
  }
};

export const disable: Story = {
  args: {
    src: '/heart.svg',
    alt: 'Cart Icon',
    width: 30,
    height: 30,
    isDisabled: true
  }
};
