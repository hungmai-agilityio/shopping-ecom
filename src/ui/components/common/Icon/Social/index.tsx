'use client';
import { Icon } from '@/ui/components';

interface socialProps {
  src: string;
  alt: string;
}
const Social = ({ src, alt }: socialProps) => {
  return (
    <Icon
      src={src}
      alt={alt}
      width={27}
      height={24}
      className="w-7 h-6"
      onClick={() => alert('Update later')}
      priority
    />
  );
};

export default Social;
