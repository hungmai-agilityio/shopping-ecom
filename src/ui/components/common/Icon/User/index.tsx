import { Icon } from '@/ui/components';

const UserIcon = () => {
  return (
    <Icon
      src="/user-white.svg"
      alt="user dropdown"
      width={24}
      height={24}
      styles="bg-primary p-2 rounded-full cursor-pointer"
    />
  );
};

export default UserIcon;
