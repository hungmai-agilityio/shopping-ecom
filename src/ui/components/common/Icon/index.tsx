import { TYPE, iconTypeClasses } from '@/constants';
import { clsx } from 'clsx';
import Image, { ImageProps } from 'next/image';

interface IconProps extends ImageProps {
  onClick?: () => void;
  isDisabled?: boolean;
  variant?: TYPE.PRIMARY | TYPE.SECOND | TYPE.THIRD;
  styles?: string;
}

const Icon = ({
  src,
  alt,
  width,
  height,
  onClick,
  isDisabled,
  variant = TYPE.PRIMARY,
  styles,
  ...props
}: IconProps) => {
  return (
    <div
      className={clsx('flex', iconTypeClasses[variant], styles, {
        'opacity-20 cursor-not-allowed': isDisabled,
        'cursor-pointer': onClick
      })}
      onClick={onClick}
    >
      <Image src={src} alt={alt} width={width} height={height} {...props} />
    </div>
  );
};

export default Icon;
