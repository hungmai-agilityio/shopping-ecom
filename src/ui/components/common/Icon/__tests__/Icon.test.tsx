import { fireEvent, render, screen } from '@testing-library/react';

// Components
import Icon from '@/ui/components/common/Icon';

describe('Icon component', () => {
  const onClickMock = jest.fn();

  const defautProps = {
    src: '/cart.svg',
    alt: 'Cart Icon',
    width: 24,
    height: 24,
    onClick: onClickMock
  };

  test('renders icon with default props', () => {
    render(<Icon {...defautProps} />);

    const icon = screen.getByAltText('Cart Icon');
    expect(icon).toBeInTheDocument();
  });

  test('calls the onClick handler when clicked', () => {
    render(<Icon {...defautProps} />);

    const icon = screen.getByAltText('Cart Icon');
    fireEvent.click(icon);

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  test('should be render match to snapshot', () => {
    const { container } = render(<Icon {...defautProps} />);

    expect(container).toMatchSnapshot();
  });
});
