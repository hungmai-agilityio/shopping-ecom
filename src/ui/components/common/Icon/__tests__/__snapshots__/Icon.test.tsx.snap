// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should be render match to snapshot 1`] = `
<div>
  <div
    class="flex cursor-pointer"
  >
    <img
      alt="Cart Icon"
      data-nimg="1"
      decoding="async"
      height="24"
      loading="lazy"
      src="/cart.svg"
      style="color: transparent;"
      width="24"
    />
  </div>
</div>
`;
